projects/libs/core-library-typescript/src/common-library/block-utils.ts

```diff
@@ -0,0 +1,38 @@
+
+import { types, logger } from ".";
+import { createHash } from "crypto";
+
+export namespace BlockUtils {
+  export function calculateBlockHash(blockComponents: {header: types.BlockHeader, body: types.BlockBody}): Buffer {
```

Comment:
Instead of a namespace + flat function, could you implement it as a class with static methods instead? It'd be more similar to what we did in other places (and what we will do in other languages).